extension.js

```diff
@@ -356,6 +364,19 @@ function update_height_setting(win) {
     settings.set_double('window-height', Math.min(1.0, current_height));
 }
 
+function window_maximize() {
+    if (!current_window)
+        return;
+
+    if (current_window.maximized_vertically) {
+        current_window.unmaximize(Meta.MaximizeFlags.VERTICAL);
+        settings.set_boolean('window-maximize', false);
```

Comment:
Why do you need to change `window-maximize` from the extension? IMO the application should control the value, and the extension should only read it (and react to changes)

Reply:
Because if you press F11 the value is set to `true`. If you then change the height the value will still be `true` even if it is not maximized.

This ensures the value is synced with the actual state of the window next time F11 is pressed.

The other option would be to set it in update window geometry I guess? (i wasn't sure which made more sense)

Reply:
`update_window_geometry()` is called in response to multiple different events, it won't be correct to set `window-maximize` to `false` there.

You probably should instead set `window-maximize` to `false` in [`unmaximize_window()`](https://github.com/amezin/gnome-shell-extension-ddterm/blob/1c9af7165a94e880d5196cea6306051321363dea/extension.js#L329). It's currently called [in response to changes in window's `maximized-vertically` property](https://github.com/amezin/gnome-shell-extension-ddterm/blob/1c9af7165a94e880d5196cea6306051321363dea/extension.js#L296) only. Setting it to `false` under `if (!win.maximized_vertically)` should be enough to keep in sync I think.

Reply:
I've opted for `update_height_settings` in the end with the view that after any event that changes the height of the window the maximize flag should be `false`. I've also tried where you say and I cannot see any difference in behavior. I can leave it there or move it to `unmaximize_window`, as you wish.